#Question Number-1
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions

**Step-1**

is there any significant difference in the diameter of the cutlet between two units

**Step-2**

Data Description

α == 0.05 (95% Confidence)

Y == Continious

X == Discrete

Is Y1 and Y2 normal ?

H0 = Y1 and Y2 are normal

H1 = Y1 and Y2 are not normal

**Normality Test**

In [3]:
import pandas as pd 
import numpy as np
from scipy import stats

In [1]:
from google.colab import files 
uploaded=files.upload()

Saving Cutlets.csv to Cutlets.csv


In [5]:
df=pd.read_csv('Cutlets.csv')
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [6]:
stats.shapiro(df['Unit A'])

ShapiroResult(statistic=0.9649459719657898, pvalue=0.31998491287231445)

P value for A == 0.32 > α

In [7]:
stats.shapiro(df['Unit B'])

ShapiroResult(statistic=0.9727305769920349, pvalue=0.5225146412849426)

P value for Unit B == 0.52 > α

Ho is accepted,that is both Y1 and Y2 are normal 

**External Condition**

External condition are same, thus we can perform paired T test

**Model**

H0 == Mean for Y1 and Y2 are equal (There is no significance difference between diameter of the Culets)

H1 == Mean for Y1 and Y2 are not equal (There is a significance difference between diameter of the Culets)


In [9]:
stats.ttest_rel(df['Unit A'],df['Unit B'])

TtestResult(statistic=0.7536787225614316, pvalue=0.4562300768038412, df=34)

P value of the paired T test is == 0.45 >α

**Thus H0 is accepted**

Mean of both Y1 nad Y2 are equal

#Question Number 2

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch

Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level

**Step-1**

Is there a significant difference in the average Turn Around Time between laboratories ?


**Step-2**

Data Description

α == 0.05 (95% Confidence)

Y == Continious

X == Discrete

Is Y1,Y2,Y3 and Y4 normal ?

H0 = Y1,Y2,Y3 and Y4 are normal

H1 = Y1,Y2,Y3 and Y4 are not normal

**Normality Test**

In [12]:
from google.colab import files
uploaded=files.upload()

Saving LabTAT.csv to LabTAT.csv


In [14]:
df_2=pd.read_csv('LabTAT.csv')
df_2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [16]:
 stats.shapiro(df_2['Laboratory 1'])

ShapiroResult(statistic=0.990183413028717, pvalue=0.5507769584655762)

P value for Laboratory 1 == 0.55 >α 



In [17]:
stats.shapiro(df_2['Laboratory 2'])

ShapiroResult(statistic=0.9936320185661316, pvalue=0.8637352585792542)

P value for Laboratory 1 == 0.86 >α 



In [18]:
stats.shapiro(df_2['Laboratory 3'])

ShapiroResult(statistic=0.988633930683136, pvalue=0.42045730352401733)

P value for Laboratory 1 == 0.42 >α 



In [19]:
stats.shapiro(df_2['Laboratory 4'])

ShapiroResult(statistic=0.9913760423660278, pvalue=0.6619619131088257)

P value for Laboratory 1 == 0.66 >α 

**H0 is accepted,that is Y1,Y2,Y3 and Y4 are normal**



**Variance**
H0 == Variance of all 4 laboratories are the same

H1 == Variance of all 4 laboratories are the not same


In [20]:
stats.levene(df_2['Laboratory 1'],df_2['Laboratory 2'],df_2['Laboratory 3'],df_2['Laboratory 4'])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

P value for Laboratory 1 == 0.51 >α 



**H0 is accepted,thus we will perform one way Anova Test**

**Model**

H0 == Mean TAT for 4 laboratories equal (There is no significance difference between TAT of the laboratories)

H1 == Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)

In [22]:
stats.stats.f_oneway(df_2['Laboratory 1'],df_2['Laboratory 2'],df_2['Laboratory 3'],df_2['Laboratory 4'])

<ipython-input-22-50fe60a9da79>:1: DeprecationWarning: Please use `f_oneway` from the `scipy.stats` namespace, the `scipy.stats.stats` namespace is deprecated.
  stats.stats.f_oneway(df_2['Laboratory 1'],df_2['Laboratory 2'],df_2['Laboratory 3'],df_2['Laboratory 4'])


F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

P vaule of the One way Anova test is == 2e-16 < α

**Thus H1 is accepted**

Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)

#Question Number 3

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions

**Step-1**

Is the male-female buyer rations are similar across regions

**Step-2**

Data Description

α == 0.05 (95% Confidence)

Y == Continious

X == Discrete

Since there are more than 2 variables we will perform chi square test

**Chi-Square Test**

In [23]:
from google.colab import files
uploaded=files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


In [25]:
df_3=pd.read_csv('BuyerRatio.csv')
df_3.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

H0 == The male-female buyer rations are similar across regions

H1 == The male-female buyer rations are not similar across regions

In [27]:
stats.chi2_contingency([df_3['East'],df_3['West'],df_3['North'],df_3['South']])

Chi2ContingencyResult(statistic=1.5959455386610577, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  442.23468701],
       [ 146.81287862, 1518.18712138],
       [ 131.11756787, 1355.88243213],
       [  72.30424052,  747.69575948]]))

P value of Chi-Square test == 0.66 > α

**H0 is accepted**

the male and female buyer rations are similar across regions

#Question Number 4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

**Step-1**

Does the defective % varies significantly by centre ?

**Step-2**

Data Description

α == 0.05 (95% Confidence)

Y == Continious

X == Discrete

Since there are more than 2 variables we will perform chi square test

** Chi-Square Test

In [29]:
from google.colab import files
uploaded=files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [31]:
df_4=pd.read_csv('Costomer+OrderForm.csv')
df_4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

H0 == The defective % does not varies by centre

H1 == The defective % does varies by centre

In [36]:
stats.chi2_contingency([df_4['Phillippines'].value_counts(),df_4['Indonesia'].value_counts(),df_4['Malta'].value_counts(),df_4['India'].value_counts()])

Chi2ContingencyResult(statistic=3.8589606858203545, pvalue=0.2771020991233144, dof=3, expected_freq=array([[271.75,  28.25],
       [271.75,  28.25],
       [271.75,  28.25],
       [271.75,  28.25]]))

P value of Chi-Square Test == 0.277 >α

**Thus H0 is accepted**

The defective % does not varies significantly by centres

